ocs_samples/basic_samples/SDS/DotNet/SdsRestApiCore/SdsRestApiCore/readme.md

```diff
@@ -50,6 +50,275 @@ replace the placeholders with the authentication-related values you received fro
+                    new StringContent(JsonConvert.SerializeObject(waveStream)));
+```
+Create and Insert Values into the Stream
+----------------------------------------
+
+A single event is a data point in the stream. An event object cannot be
+empty and should have at least the key value of the SDS type for the
+event.  First the event is created locally by instantiating a new WaveData 
+object:
+```C#
+return new WaveData
+            {
+                Order = order,
+                Radians = radians,
+                Tau = radians / (2 * Math.PI),
+                Sin = multiplier * Math.Sin(radians),
+                Cos = multiplier * Math.Cos(radians),
+                Tan = multiplier * Math.Tan(radians),
+                Sinh = multiplier * Math.Sinh(radians),
+                Cosh = multiplier * Math.Cosh(radians),
+                Tanh = multiplier * Math.Tanh(radians),
+            };
+```
+Then submit the event using the proper API call:
+```C#
+response = await httpClient.PostAsync($"api/{apiVersion}/Tenants/{tenantId}/Namespaces/{namespaceId}/Streams/{waveStream.Id}/Data",
+                    new StringContent(JsonConvert.SerializeObject(singleWaveList)));
+```
+Similarly, we can build a list of objects and insert them in bulk by sending multiple waves in the API call:
+```C#
+List<WaveData> waves = new List<WaveData>();
+                for (int i = 2; i < 20; i += 2)
+                {
+                    WaveData newEvent = GetWave(i, 2, 2.0);
+                    waves.Add(newEvent);
+                }
+response = await httpClient.PostAsync($"api/{apiVersion}/Tenants/{tenantId}/Namespaces/{namespaceId}/Streams/{waveStream.Id}/Data",
+                    new StringContent(JsonConvert.SerializeObject(waves)));
+```
+Retrieve Values from a Stream
+-----------------------------
+
+There are many methods in the SDS REST API allowing for the retrieval of
+events from a stream. The retrieval methods take string type start and
+end values; in our case, these are the start and end ordinal indices
+expressed as strings. The index values must
+capable of conversion to the type of the index assigned in the SdsType.
+```C#
+response = await httpClient.GetAsync($"api/{apiVersion}/Tenants/{tenantId}/Namespaces/{namespaceId}/Streams/{waveStream.Id}/Data?startIndex=0&endIndex={waves[waves.Count - 1].Order}");
+List<WaveData> retrievedList = JsonConvert.DeserializeObject<List<WaveData>>(await response.Content.ReadAsStringAsync());
+```
+You can retreive the values in the form of a table (in this case with headers)
+```C#
+response = await httpClient.GetAsync($"api/{apiVersion}/Tenants/{tenantId}/Namespaces/{namespaceId}/Streams/{waveStream.Id}/Data?startIndex=0&endIndex={waves[waves.Count - 1].Order}&form=tableh");
+```
+You can retrieve a sample of your data to show the overall 
+trend. In addition to the start and end index, we also 
+provide the number of intervals and a sampleBy argument. Intervals 
```

Comment:
I think it's parameter not argument when it's part of URL... Also 'intervals determines' doesn't sound right grammatically right to me but that happens in documentation so you can take or leave this on but I would say 'Intervals parameter determines...'